### DISCUSSION TITLE
微调PP-OCRv3文本识别模型，对图片竖着文字识别效果差，训练数据集需要特别处理吗？

### DISCUSSION THREAD

**Discussion by zxy2020** [PaddlePaddle/PaddleOCR#12208] [Q&A] [answered]
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/59002646/68a3d768-c3e7-4d10-bab3-376b4d962ad6)
```
python tools/train.py -c configs/rec/PP-OCRv3/ch_PP-OCRv3_rec.yml -o \
    Global.pretrained_model=ch_PP-OCRv3_rec_train/student.pdparams \
    Global.save_model_dir=./output/rec_ppocr_v3_exp2 \
		Global.epoch_num=200 \
		Global.checkpoints=output/rec_ppocr_v3/best_accuracy.pdparams \
    Global.eval_batch_step="[0, 90]" \
    Optimizer.lr.name=Const \
    Optimizer.lr.learning_rate=0.0005 \
    Optimizer.lr.warmup_epoch=3 \
    Train.dataset.data_dir=./ \
    Train.dataset.label_file_list=[./train_data/rec/rec_gt_train.txt] \
    Eval.dataset.data_dir=./ \
    Eval.dataset.label_file_list=[./train_data/rec/rec_gt_val.txt]

**Comment by rizwan777** [comment]
yes !! we definitely need to consider alignment of text.

**Comment by zxy2020** [comment]
> yes !! we definitely need to consider alignment of text.


thanks you reply! could you tell me the url of doc about alignment of text?
谢谢你的回复，请问哪里有关于文本对齐的文档或者说明？

**Comment by rizwan777** [comment, marked as answer]
actually i dont have that in hand.

but i have worked before in similar project that i used to preprocess the image (text).

Preprocessing:
Depending on the layout of the Chinese text (vertical or horizontal), you might need to perform additional preprocessing steps, such as rotating the image if the text is vertical. OpenCV provides functions to perform image rotations.


# Rotate the image clockwise by 90 degrees
rotated_image = cv2.rotate(image, cv2.ROTATE_90_CLOCKWISE)

**Comment by zxy2020** [comment]
> 

thank you !  I find the  code   at   [infer process](https://github.com/PaddlePaddle/PaddleOCR/blob/0850586667308d38e113447e8d095e955092fe53/tools/infer/utility.py#L629C36-L629C36)
```
    dst_img_height, dst_img_width = dst_img.shape[0:2]
    if dst_img_height * 1.0 / dst_img_width >= 1.5:
        dst_img = np.rot90(dst_img)
    return dst_img
